src/NTDll.Desktop/NTDll+UNICODE_STRING.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.Runtime.InteropServices;
+
+#pragma warning disable SA1401 // Fields must be private
+
+    /// <content>
+    /// Contains the <see cref="UNICODE_STRING"/> nested struct.
+    /// </content>
+    public static partial class NTDll
+    {
+        /// <summary>
+        /// The UNICODE_STRING structure is used to define Unicode strings.
+        /// </summary>
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Unicode)]
+        public struct UNICODE_STRING
+        {
+            /// <summary>
+            /// The length, in bytes, of the string stored in <see cref="Buffer"/>.
+            /// </summary>
+            public ushort Length;
+
+            /// <summary>
+            /// The length, in bytes, of <see cref="Buffer"/>.
+            /// </summary>
+            public ushort MaximumLength;
+
+            /// <summary>
+            /// Pointer to a buffer used to contain a string of wide characters.
+            /// </summary>
+            [MarshalAs(UnmanagedType.LPWStr)]
+            public string Buffer;
```

Comment:
I think we should remove `MarshalAs` and change the type to `char*`, since [`LPWStr` should be null-terminated strings](https://msdn.microsoft.com/en-us/library/s9ts558h(v=vs.110).aspx#cpcondefaultmarshalingforstringsanchor5) and [the docs for this struct](https://msdn.microsoft.com/en-us/library/windows/hardware/ff564879(v=vs.85).aspx) (if I'm looking at the right definition) doesn't guarantee that it's null terminated.